earth_enterprise/src/earthplugin/SConscript

```diff
@@ -20,47 +20,23 @@ import os
 
 Import('env')
 
-# Get Fusion Version (use same version for server).
-# Based on Andrei's code in fusion/portableglobe/SConscript.
-def VersionNumber(source):
-  """Reads the Fusion Version number."""
-  try:
-    fp = open(source, 'r')
-    line = fp.readline()
-    while len(line) != 0:
-      if line[0] != '#':
-        version = line.strip()
-        if len(version):
-          return version
-      line = fp.readline()
-    print 'Unable to find Fusion Version Number.'
-    Exit(1)
-  except Exception, exp:
-    print 'Unable to find Fusion Version Number: ', exp
-
-def WriteBuildInfoFile(build_info, version_file_path):
-  try:
-    fp = open(version_file_path, 'w')
-    fp.write(build_info)
-    fp.close()
-  except Exception, exp:
-    print 'Unable to write Build Info File: ', exp
-
-fusion_version_basic_file_path = File('#fusion_version.txt').abspath
 server_version_file_dir = Dir('.').abspath
 
 # Create 'earthplugin' directory if it doesn't exist.
 if not os.path.exists(server_version_file_dir):
   os.mkdir(server_version_file_dir)
 
-server_version_file_path = File('%s/%s' %(server_version_file_dir, 'version.txt')).abspath
-version_basic = VersionNumber(fusion_version_basic_file_path)
-build_info = ('%s-%s' % (version_basic, time.strftime('%Y%m%d')))
-WriteBuildInfoFile(build_info, server_version_file_path)
-
 # Copy files from '../earth_enterprise/src/earthplugin/..' directory.
 env.installDirExcluding('earth', Dir('#earthplugin').abspath, ['SConscript'])
 
 # Copy files from '../earth_enterprise/src/{REL_XXX|OPT_XXX}/earthplugin/..'
-# directory. There is only one file 'version.txt' in this folder.
-env.installDirExcluding('earth', '.', [])
+# directory.
+
+# contains the build date in YYYY-MM-DD format
+env.install('earth', File('../builddate.txt').abspath)
```

Comment:
A couple of these files exist purely to get the needed information into the intro web pages.  And a couple are because the existing code wants the data available as C++ macros.

A general "this is the build information" config file would be cleaner, though. It will naturally add the need for additional parsing calls in various places, though.

Reply:
Well, if the version is a string, you have to parse it to unpack the various parts of it.  It's not very complicated parsing, but there are serializers and deserializers in both Python and Groovy so you can put a data structure in a file, e.g. from Python, and read it back, e.g. in Groovy.  You could expand a template to write a Python structure as a C++ class, or generate C macros.

What kind of expansion is used to get it into Web pages?

Reply:
For the web pages, there's a small javascript routine that just grabs the contents of the file, and appends it to the appropriate text box.

Reply:
Well, Java Script is pretty good reading JSON. :stuck_out_tongue:
And I think it'd be fine with other formats, as well.